### ISSUE TITLE
MoarVM 2020.12 does not build on Fedora rawhide (while it builds on 31, 32, 33)

### ISSUE THREAD

**Issue by nxadm** [MoarVM/MoarVM#1425]
Hi,
Fedora rawhide is the development branch of Fedora Linux and it will become the next stable branch. MoarVM can not be built, while with identical sources and OS packages it works fine on other supported Fedora releases. The error when running make is this:
```make
linking 3rdparty/libatomicops/src/libatomic_ops.a
configure: error: cannot find sources (src/atomic_ops.c) in . or ..
make: *** [Makefile:733: 3rdparty/libatomicops/src/libatomic_ops.a] Error 1
Error: Process completed with exit code 2.
```

While the file is there (@JJ++) :
```
# find /moarvm -name "atomic_ops.c"
/moarvm/3rdparty/libatomicops/src/atomic_ops.c
```

Full OS setup and error here: https://github.com/nxadm/rakudo-pkg/runs/1794747938?check_suite_focus=true#step:5:306

The weirdness of the error would suggest a bug in binutils except that Fedora rawhide and 33 have the same version.

To reproduce locally (on Linux) with a container (you can replace [podman](https://podman.io/getting-started/installation) with docker, podman does not need root privileges):
```
podman run -ti fedora:rawhide
```

In the container:
```
dnf -q -y upgrade                  
dnf -q -y groupinstall 'Development Tools'                             
dnf -q -y install libzstd-devel perl-core 
export INSTALL_ROOT=/opt/rakudo-pkg CONFIG_SHELL=/bin/bash MOARVM_VERSION=2020.12
curl -sSL -o moarvm.tar.gz https://github.com/MoarVM/MoarVM/releases/download/$MOARVM_VERSION/MoarVM-${MOARVM_VERSION}.tar.gz
mkdir moarvm && tar xzf moarvm.tar.gz -C moarvm --strip-components=1
cd moarvm
perl ./Configure.pl --prefix=$INSTALL_ROOT --relocatable
make
```

Related to #1424.

**Comment by nxadm** [comment]
Weirder: running the same container version with a "debug" configure builds fine (same on #1424):

```
perl ./Configure.pl --prefix=$INSTALL_ROOT --relocatable  --debug --optimize=0
```

**Comment by salortiz** [comment]
Can't reproduce using the image at https://dl.fedoraproject.org/pub/fedora/linux/development/rawhide/Cloud/x86_64/images/Fedora-Cloud-Base-Rawhide-20210201.n.1.x86_64.raw.xz in a fresh `systemd-nspawn` container.

In the container same recipe as yours:

```
…
linking 3rdparty/libuv/libuv.a
linking libmoar.so
linking moar
/usr/bin/perl build/mk-moar-pc.pl pkgconfig/moar.pc
[root@Fedora-Cloud-Base-Rawhide-20210201 moarvm]# make install
…
[root@Fedora-Cloud-Base-Rawhide-20210201 moarvm]# cd /opt/rakudo-pkg/bin/
[root@Fedora-Cloud-Base-Rawhide-20210201 bin]# ./moar --version
This is MoarVM version 2020.12 built with JIT support
```

**Comment by nxadm** [comment]
@salortiz: I tried on github actions (ubuntu 18.04 VM + docker) and with podman on Centos 7. I'll run the builds again tomorrow.

**Comment by nxadm** [comment]
Podman on CentOS 8:
```
building dyncall...
Configuration written to Makefile.config
ar: creating libdyncall_s.a
ar: creating libdyncallback_s.a
ar: creating libdynload_s.a
done.
linking 3rdparty/libatomicops/src/libatomic_ops.a
configure: error: cannot find sources (src/atomic_ops.c) in . or ..
make: *** [Makefile:733: 3rdparty/libatomicops/src/libatomic_ops.a] Error 1
```

**Comment by nxadm** [comment]
@salortiz Can you test with docker and/or podman with the dockerhub image fedora:rawhide?

**Comment by salortiz** [comment]
@nxadm
Tested with podman in my F33, the build succeeds:
```
$ podman run -ti fedora:rawhide
```
Then
```
[root@d7ef6a0ce112 ~]# dnf -q -y groupinstall 'Development Tools'
[root@d7ef6a0ce112 ~]# dnf -q -y install libzstd-devel perl-core
…
[root@d7ef6a0ce112 ~]# cd moarvm; perl ./Configure.pl --prefix=$INSTALL_ROOT --relocatable
Welcome to MoarVM!
…
[root@d7ef6a0ce112 moarvm]# make
…
building dyncall...
Configuration written to Makefile.config
ar: creating libdyncall_s.a
ar: creating libdyncallback_s.a
ar: creating libdynload_s.a
done.
linking 3rdparty/libatomicops/src/libatomic_ops.a
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
…
make[1]: Entering directory '/root/moarvm/3rdparty/libatomicops/src'
make  all-am
make[2]: Entering directory '/root/moarvm/3rdparty/libatomicops/src'
gcc -DHAVE_CONFIG_H   -I../src -I../src  -fPIC -Wall -Wextra -std=gnu99 -Wextra -Wall -Wno-unused-parameter -Wno-unused-function -Wno-missing-braces -Werror=pointer-arith -O3 -DNDEBUG  -D_REENTRANT -D_FILE_OFFSET_BITS=64 -fPIC -DMVM_HEAPSNAPSHOT_FORMAT=3 -DMVM_TRACING=0 -DMVM_CGOTO=1 -DMVM_RDTSCP=1 -MT atomic_ops.o -MD -MP -MF .deps/atomic_ops.Tpo -c -o atomic_ops.o atomic_ops.c
mv -f .deps/atomic_ops.Tpo .deps/atomic_ops.Po
rm -f libatomic_ops.a
ar cru libatomic_ops.a atomic_ops.o 
ranlib libatomic_ops.a
make[2]: Leaving directory '/root/moarvm/3rdparty/libatomicops/src'
…
linking 3rdparty/libuv/libuv.a
linking libmoar.so
linking moar
/usr/bin/perl build/mk-moar-pc.pl pkgconfig/moar.pc
[root@d7ef6a0ce112 moarvm]#
```
Note that using `fedora:rawhide` the image used isn't from dockerhub:
```
$ podman images
REPOSITORY                         TAG      IMAGE ID      CREATED       SIZE
registry.fedoraproject.org/fedora  rawhide  2506f6301164  10 hours ago  188 MB
```

**Comment by nxadm** [comment]
@salortiz OK, I changed the way how I retrieved the sources. This does not explain why previous versions don't fail on my setup. I will keep you posted.

**Comment by nxadm** [comment]
I think that the explanation is that the tar from https://github.com/MoarVM/MoarVM/tarball/$TAG does not include the submodules, but the Configure.pl script does not care as long git is installed.

**Comment by nxadm** [comment]
Merged with #1424
